### PR TITLE
Refactor show_log_panel

### PR DESCRIPTION
**Pull request by randy3k**
Now it supports on_highlight(entry) and is used in GsBlameCommand.
It also fixes a bug when canceling the panel of `Pick a new commit to blame`

-----

The maintainers of this repo require that all pull request submitters agree and adhere to the following:

- [x] I have read and agree to the [contribution guidelines](https://github.com/divmain/GitSavvy/blob/master/CONTRIBUTING.md).
 _(required)_
- [x] All related documentation has been updated to reflect the changes made. _(required)_
- [x] My commit messages are cleaned up and ready to merge. _(required)_

The maintainers of this repository require you to select the semantic version type that
the changes in this pull request represent.  Please select one of the following:
- [ ]  major
- [ ]  minor
- [x]  patch
- [ ]  documentation only

### CODE REVIEW

**Comment by randy3k** [comment]
@stoivo free to review? I remember you are on vacation though.

File: core/ui_mixins/quick_panel.py [hunk 1]
```diff
@@ -139,24 +142,15 @@ def show_paginated_panel(items, on_done, flags=None, selected_index=None, on_hig
...
-
-def show_log_panel(entries, on_done, limit=6000, selected_index=None):
-    """
-    Display log entries in quick panel with pagination, and execute on_done(commit)
-    when item is selected. `entries` can be either a list or a generator of LogEnty.
-
-    """
-    lp = LogPanel(entries, on_done, limit=limit, selected_index=selected_index)
-    lp.show()
+    pp
```
Comment (asfaltboy) [inline core/ui_mixins/quick_panel.py]: typo ?

File: core/ui_mixins/quick_panel.py [hunk 2]
```diff
@@ -232,6 +239,22 @@ def on_selection(self, index):
...
+
+    """
+    lp = LogPanel(
+        entries,
+        on_done,
+        limit=limit,
+        selected_index=selected_index,
+        on_highlight=on_highlight)
+    lp.show()
+    lp
```
Comment (asfaltboy) [inline core/ui_mixins/quick_panel.py]: typo again ?

File: core/ui_mixins/quick_panel.py [hunk 3]
```diff
@@ -168,7 +162,7 @@ def __init__(self, items, on_done, **kwargs):
         self.on_done = on_done
         for option in ['flags', 'selected_index', 'on_highlight',
                        'limit', 'next_message', ]:
-            if option in kwargs:
+            if option in kwargs and kwargs[option]:
```
Comment (asfaltboy) [inline core/ui_mixins/quick_panel.py]: why add `and kwargs[option]` ? won't this effectively skip any "falsy" value?

E.g doing: `PaginatedPanel(selected_index=0, limit=0)` would ignore both values and keep them at the default `None`.

File: core/ui_mixins/quick_panel.py (hunk 1 above)
Comment (randy3k) [inline core/ui_mixins/quick_panel.py]: I meant `return pp`. Doing julia too much earlier. In julia, the last expression always returns

File: core/ui_mixins/quick_panel.py (hunk 2 above)
Comment (randy3k) [inline core/ui_mixins/quick_panel.py]: I meant `return lp`.

File: core/ui_mixins/quick_panel.py (hunk 3 above)
Comment (randy3k) [inline core/ui_mixins/quick_panel.py]: right, it should be `kwargs[option] is not None`. If the nullness is not checked, it may overwrite the default methods, e.g `on_highlight`.

File: core/commands/blame.py [hunk 4]
```diff
@@ -376,52 +378,33 @@ def show_file_at_commit(self, from_line=False):
...
 
+
 class GsBlamePickCommitCommand(TextCommand, GitCommand):
 
     def run(self, *args, commit_hash=None):
         sublime.set_timeout_async(lambda: self.run_async(self.file_path), 0)
 
     def run_async(self, file_path):
-        settings = self.view.settings()
-        settings.set("git_savvy.commit_hash_old", settings.get("git_savvy.commit_hash"))
```
Comment (stoivo) [inline core/commands/blame.py]: Where did this go?

Comment (randy3k) [inline core/commands/blame.py]: if the panel is cancelled, the commit will set to `self.commit_hash`, which is the original commit (line 407).

File: core/ui_mixins/quick_panel.py (hunk 3 above)
Comment (randy3k) [inline core/ui_mixins/quick_panel.py]: It seems that your comment has been discarded. The reason that we need to check the nullness is the usage of LogPanel. LogPanel has a default on_highlight method. If the passing argument is not checked for beingNone, the None value would override the default.

Comment (asfaltboy) [inline core/ui_mixins/quick_panel.py]: Oh I think I see now.. we do this so that show_log_panel can accept `on_highlight`?

I truly found it a bit hard to follow, maybe we need to add some comments or docstrings, at least so we don't forget why it's done in the future?

File: common/util/view.py [hunk 5]
```diff
@@ -68,10 +68,6 @@ def refresh_gitsavvy(view, refresh_sidebar=False, refresh_status_bar=True):
 
     if view.settings().get("git_savvy.interface") is not None:
         view.run_command("gs_interface_refresh")
-    if view.settings().get("git_savvy.branch_commit_history_view") is not None:
-        view.run_command("gs_branches_diff_commit_history_refresh")
```
Comment (randy3k) [inline common/util/view.py]: and `branch_commit_history_view` no longer exists.

**Comment by randy3k** [comment]
Just want to let you guys know that I am working of a refactoring of the Blame walking and hoping to bring some new features, for example

1. syntax highlight
2. line forwarding

File: core/ui_mixins/quick_panel.py (hunk 3 above)
Comment (randy3k) [inline core/ui_mixins/quick_panel.py]: comment added.

**Comment by randy3k** [comment]
@asfaltboy does it look good now?
